scalac-scoverage-runtime/shared/src/main/scala/scoverage/Invoker.scala

```diff
@@ -31,18 +35,27 @@ object Invoker {
     // times since for coverage we only care about 1 or more, (it just slows things down to
     // do it more than once), anything we can do to help is good. This helps especially with code
     // that is executed many times quickly, eg tight loops.
-    if (!ids.contains(dataDir, id)) {
+    if (!dataDirToIds.contains(dataDir)) {
+      // Guard against SI-7943: "TrieMap method getOrElseUpdate is not thread-safe".
```

Comment:
I noticed that this was also calling `getOrElseUpdate` on a `TrieMap` down in `files.getOrElseUpdate`, but that case wasn't an issue because that map is used only in a single thread (since it comes from a non-inheritable `ThreadLocal`).

To help make this clearer in the code, I went ahead and used a plain `mutable.HashMap` at that other site.